etna/packages/etna-js/plots/components/category_plot/category_plot.jsx

```diff
@@ -0,0 +1,130 @@
+import React, {Component} from 'react';
+import {flatten} from '../../../utils/types';
+import * as d3 from 'd3';
+import Bars from './bars';
+import StackedBars from './stacked_bars';
+import Boxes from './boxes';
+import Swarms from './swarms';
+import PlotCanvas from '../plot_canvas';
+import {empty, validDomain} from '../../utils/plot';
+
+const COMPONENTS = {
+  bar: Bars,
+  stackedbar: StackedBars,
+  box: Boxes,
+  swarm: Swarms
+};
+
+export const categoryGroups = (category, value, label, groupFunc) => {
+  let category_names = [...new Set(category.values)];
+
+  return category_names.map((category_name, index) => {
+    let indexes = category
+      .which((c) => c == category_name)
+      .filter((i) => i != null);
+    let category_values = value(indexes);
+    let category_labels = label ? label(indexes) : [];
+
+    let vl = category_values
+      .map((_, i) =>
+        category_values[i] != null
+          ? [category_values[i], category_labels[i]]
+          : null
+      )
+      .filter((_) => _)
+      .sort((a, b) => a[0] - b[0]);
+
+    category_values = vl.map(([v, l]) => v);
+    category_labels = vl.map(([v, l]) => l);
+
+    return groupFunc(category_name, category_values, category_labels);
+  });
+};
+
+const SeriesComponent = ({
+  series,
+  index,
+  count,
+  gutter,
+  gap,
+  xScale,
+  ...props
+}) => {
+  let Component = COMPONENTS[series.series_type];
```

Comment:
super minor stye nit, but we could use an eager return to remove the need for the `Component &&` and any cascading nil checks.:

```
let Component = 
if (Component == null) return null;
// rest of the code can assume it is non null
```

Reply:
Sure, I can take a look at that!